tools/eslint/src/check-license.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright 2024 Google Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const licenseHeader = `/**
+ * @license
+ * Copyright 2024 Google Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */\n\n`;
+
+/**
+ * @type import("eslint").Rule.RuleModule
+ */
+
+const enforceLicenseRule = {
+  // module.exports = {
+  meta: {
+    type: 'layout',
+    docs: {
+      description: 'Validate existence of license header',
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      licenseRule: 'Add license header.',
+    },
+  },
+  create(context) {
+    const sourceCode = context.sourceCode;
+    const comments = sourceCode.getAllComments();
+    const header = comments[0];
+
+    return {
+      Program(node) {
+        if (header?.value.includes('@license')) {
+          return;
+        }
+
+        // Add header license
+        if (!header) {
+          context.report({
```

Comment:
I think we can use this for both case of `No Header` and `Header without @license`, as most likely we will still update/review it manually.

Reply:
I merged the cases together. 
The two cases had conflicting fixes, and if I understood your comment correct I have omitted the fixes as well.

Reply:
Sorry about this, I though we can have the same logic for both, it seems like EsLint does not like it when you do that so we should split them as you have suggested.

Reply:
Sorry about this as well, I believe my updated fixer accounts for both cases as you mentioned